lib/spack/spack/package.py

```diff
@@ -1816,13 +1829,10 @@ def do_test(self, dirty=False):
                 return
 
         # Clear test failures
-        self.test_failures = []
-        self.test_log_file = self.test_suite.log_file_for_spec(self.spec)
-        self.tested_file = self.test_suite.tested_file_for_spec(self.spec)
-        fsys.touch(self.test_log_file)  # Otherwise log_parse complains
-
-        kwargs = {'dirty': dirty, 'fake': False, 'context': 'test'}
-        spack.build_environment.start_build_process(self, test_process, kwargs)
+        with self._setup_test():
+            kwargs = {'dirty': dirty, 'fake': False, 'context': 'test'}
+            spack.build_environment.start_build_process(
+                self, test_process, kwargs)
```

Comment:
I think if you push the context manager inside the test_process method, then you can include error handling in the context manager instead of having to handle it separately at the end of `install_time_callbacks`

Reply:
The flow in `test_process` differs a bit from `_run_default_install_time_test_callbacks` in terms of options and flagging test state.

Are you also suggesting embedding use of the `log_output` context manager in `_setup_test()`?

Reply:
Nvm.  I pushed it too anyway.  Provides consistent log output (see updated description).